Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1687,7 +1695,46 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
                 allReportedFileAccesses: allFileAccesses,
                 detouringStatuses: result.DetouringStatuses,
                 maxDetoursHeapSize: maxDetoursHeapSize,
-                containerConfiguration: m_containerConfiguration);
+                containerConfiguration: m_containerConfiguration,
+                pipProperties: pipProperties);
+        }
+
+        private async Task<Dictionary<string, int>> UpdatePipPropertiesCountsAsync(SandboxedProcessResult result, Dictionary<string, int> pipProperties)

```

Comment:
>private async Task<Dictionary<string, int>> [](start = 8, length = 43)

I think we can have just `private async Task Update...`; no need to return a dictionary. Or this method will create a new instance of Dictionary, and thus you don't have to pass it as an argument.